cdap-watchdog/src/main/java/io/cdap/cdap/logging/appender/ExtensionLogAppender.java

```diff
@@ -0,0 +1,110 @@
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.logging.appender;
+
+import com.google.inject.Inject;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.logging.appender.loader.LogPublisherExtensionLoader;
+import io.cdap.cdap.logs.LogPublisher;
+import java.io.IOException;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A log appender that delegates log messages to a dynamically loaded log publisher.
+ */
+public class ExtensionLogAppender extends LogAppender {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ExtensionLogAppender.class);
+  private static final String APPENDER_NAME = ExtensionLogAppender.class.getName();
+
+  private final String provider;
+  private final CConfiguration cConf;
+  private LogPublisher publisher;
+
+  /**
+   * Constructs an ExtensionLogAppender with the given configuration.
+   *
+   * @param cConf The configuration object containing the properties to configure the appender.
+   */
+  @Inject
+  public ExtensionLogAppender(CConfiguration cConf) {
+    this.cConf = cConf;
+    this.provider = cConf.get(Constants.Logging.LOG_PUBLISHER_PROVIDER);
+    setName(APPENDER_NAME);
+  }
+
+  /**
+   * Starts the appender and initializes the log publisher.
+   */
+  @Override
+  public void start() {
+    publisher = loadLogPublisher();
+    Optional.ofNullable(this.publisher)
+        .ifPresent(
+            publisher -> publisher.initialize(new DefaultLogPublisherContext(cConf, provider)));
+    super.start();
+    LOG.info("Successfully started {}", APPENDER_NAME);
+  }
+
+  /**
+   * Stops the appender and releases resources associated with the log publisher.
+   */
+  @Override
+  public void stop() {
+    super.stop();
+
+    Optional.ofNullable(this.publisher)
+        .ifPresent(publisher -> {
+          try {
+            publisher.close();
+            LOG.info("Successfully stopped {}", APPENDER_NAME);
+          } catch (IOException e) {
+            LOG.warn("Failed to stop {}", APPENDER_NAME, e);
+          }
+        });
+  }
+
+  /**
+   * Appends a log message by delegating to the log publisher.
+   *
+   * @param logMessage The log message to append.
+   */
+  @Override
+  protected void appendEvent(LogMessage logMessage) {
+    logMessage.prepareForDeferredProcessing();
+    logMessage.getCallerData();
+    Optional.ofNullable(this.publisher)
+        .ifPresent(
+            publisher -> publisher.publish(logMessage));
+  }
+
+  /**
+   * Loads the log publisher using the extension loader.
+   *
+   * @return the {@link LogPublisher} instance
+   */
+  private LogPublisher loadLogPublisher() {
+    LogPublisher publisher = new LogPublisherExtensionLoader(cConf).get(provider);
+    if (publisher == null) {
+      LOG.info("Failed to load log publisher: {}", provider);
```

Comment:
So, why do you want to continue here? Log absence is a pretty serious problem. Inability to load a publisher is a misconfiguration that should be fixed, not a transient issue.
The only case where it's OK is when it's explicitly disabled, so you need to differentiate those. I would also probably either create a NopPublisher or at least make the field `Optional<Nullable>` to save calling `ofNullable` each time.